test/ember.test.js

```diff
@@ -0,0 +1,135 @@
+    }
+
+    const hash = superRentalsHashes['ember-source@~3.27'];
+    pathToSuperRentals = join(fileURLToPath(tmpDir), `super-rentals-${hash}`);
+
+    if (!existsSync(pathToSuperRentals)) {
+      // Download archive
+      const archiveURL = `https://github.com/ember-learn/super-rentals/archive/${hash}.zip`;
+
+      commandSync(`wget ${archiveURL}`, {
+        cwd: tmpDir,
+        shell: true
+      });
+      commandSync(`unzip ${hash}.zip`, { cwd: tmpDir, shell: true });
+      commandSync(`rm ${hash}.zip`, { cwd: tmpDir, shell: true });
+      commandSync('pnpm install', { cwd: pathToSuperRentals });
+    }
+  });
+
+  beforeEach(async () => {
+    project = Project.fromDir(pathToSuperRentals, { linkDeps: true, linkDevDeps: true });
+
+    project.addDevDependency('ember-source', '~3.28.0');
+    project.addDevDependency('ember-cli', '~3.28.0');
+
+    project.mergeFiles({
+      app: {
+        components: {
+          'share-button.js': `import { inject as service } from '@ember/service';
+    import Component from '@glimmer/component';
+
+    const TWEET_INTENT = 'https://twitter.com/intent/tweet';
+
+    export default class ShareButtonComponent extends Component {
+      @service router;
+      @service locale;
+
+      get currentURL() {
+        return new URL(this.router.currentURL, window.location.origin);
+      }
+
+      get shareURL() {
+        let url = new URL(TWEET_INTENT);
+
+        url.searchParams.set('url', this.currentURL);
+
+        if (this.args.text) {
+          url.searchParams.set('text', this.args.text);
+        }
+
+        if (this.args.hashtags) {
+          url.searchParams.set('hashtags', this.args.hashtags);
+        }
+
+        if (this.args.via) {
+          url.searchParams.set('via', this.args.via);
+        }
+
+        url.searchParams.set('locale', this.locale.current());
+
+        return url;
+      }
+    }`
+        },
+        services: {
+          'locale.js': `
+            import Service from '@ember/service';
+
+            export default class LocaleService extends Service {
+                current() {
+                    return 'en-US';
+                }
+            }
+          `
+        }
+      }
+    });
+
+    // Setup project with dependencies to use rehearsal e.g. typescript, eslint, prettier, @glint/*
+    await setupEmberProject(project);
+
+    const bin = resolveCLIBin(project);
+    // Set up command for tests
+    run = runCommandFactory(bin, { cwd: project.baseDir });
+  });
+
+  afterEach(() => {
+    project.dispose();
+  });
+
+  test('graph', () => {
+    const results = run(['graph']);
+
+    expect(results.exitCode).toBe(0);
+    expect(results.stdout, 'should show ordering between locale service and share-button')
+      .toContain(`[DATA] app/services/locale.js
+[DATA] app/components/share-button.js`);
+
+    expect(results.stdout).toMatchSnapshot();
+  });
+
+  test('migrate', () => {
+    const results = run(['migrate', '--ci']);
```

Comment:
we need to include the snapshots of file transforms with type inference

Reply:
With super-rentals as it stands, no types get inferred/added only TODOs only. The last update asserts that the files have been commented with rehearsal TODOs.